pkg/db/db.go

```diff
@@ -0,0 +1,51 @@
+package db
+
+import (
+	"database/sql"
+	"os"
+
+	_ "modernc.org/sqlite"
+)
+
+// DB хранит активное соединение с базой данных.
+var DB *sql.DB
+
+// Схема для создания таблицы задач
+const schema = `
+CREATE TABLE IF NOT EXISTS scheduler (
+	id INTEGER PRIMARY KEY AUTOINCREMENT,
+	date CHAR(8) NOT NULL,
+	title VARCHAR(255) NOT NULL,
+	comment TEXT,
+	repeat VARCHAR(128) NOT NULL
+);
+CREATE INDEX IF NOT EXISTS date_idx ON scheduler (date);
+`
+
+// Init инициализирует соединение с БД.
+func Init() error {
+	var dbFile string
+	// Проверяем, запущено ли приложение в Docker.
+	if os.Getenv("RUNNING_IN_DOCKER") == "true" {
+		// Используем in-memory базу данных.
+		dbFile = "file::memory:?cache=shared"
+	} else {
+		// Используем файловую базу данных.
+		dbFile = os.Getenv("TODO_DBFILE")
+		if dbFile == "" {
+			dbFile = "scheduler.db"
+		}
+	}
+
+	var err error
+	DB, err = sql.Open("sqlite", dbFile)
+	if err != nil {
+		return err
```

Comment:
Лучше всегда оборачивать ошибки
```suggestion
		return fmt.Errorf("error opening database: %w", err)
```